ledfx/virtuals.py

```diff
@@ -874,8 +825,15 @@ def flush(self, pixels=None):
         if pixels is None:
             pixels = self.assembled_frame
 
-        if self._os_active:
-            self.oneshot_weight()
+        # Where we update oneshots
```

Comment:
Would this be better as

for oneshot in self.oneshot:

Oh it's got remove in the loop, you have to be VERY careful with that, can't remember what the safest way to deal with that is...

Reply:
Yep I followed the standard way of removing from a list why iterating through it found here:

https://www.geeksforgeeks.org/remove-items-from-a-list-while-iterating-python/

So it should be all good. This was also one of the ways coderabbitai suggested to do it.

Reply:
ok, if you are following a pattern, great, it is so easy to pull the rug out from the structure when removing otherwise